process_report/tests/unit/processors/test_coldfront_fetch_processor.py

```diff
@@ -0,0 +1,92 @@
+from unittest import TestCase, mock
+import pandas
+
+from process_report.tests import util as test_utils
+
+
+class TestColdfrontFetchProcessor(TestCase):
+    def _get_test_invoice(
+        self,
+        project,
+        pi=None,
+        institute_code=None,
+    ):
+        if not pi:
+            pi = [""] * len(project)
+
+        if not institute_code:
+            institute_code = [""] * len(project)
+
+        return pandas.DataFrame(
+            {
+                "Manager (PI)": pi,
+                "Project - Allocation": project,
+                "Institution - Specific Code": institute_code,
+            }
+        )
+
+    def _get_mock_allocation_data(self, project_list, pi_list, institute_code_list):
+        mock_data = {}
+        for i, project in enumerate(project_list):
+            mock_data[project] = {}
+            mock_data[project]["project"] = {"pi": pi_list[i]}
+            mock_data[project]["attributes"] = {
+                "Institution-Specific Code": institute_code_list[i]
+            }
+
+        return mock_data
+
+    @mock.patch(
+        "process_report.processors.coldfront_fetch_processor.ColdfrontFetchProcessor._fetch_coldfront_allocation_api",
+    )
+    def test_coldfront_fetch(self, mock_get_allocation_data):
+        mock_get_allocation_data.return_value = self._get_mock_allocation_data(
+            ["P1", "P2", "P3", "P4"],
+            ["PI1", "PI1", "", "PI12"],
+            ["IC1", "", "", "IC2"],
```

Comment:
Just so I understand: you are associating `PI1` with Project `P2` but then it doesn't have an Institution Code. Is the institution code associated with the Project or the PI? Probably doesn't matter for the test, but I would like it to match the real world data.

Reply:
From the invoicing requirements file, the institute code is unique to each project, so I guess this test data could be real world data.